File: src/main/java/org/fountainmc/api/command/Command.java
package org.fountainmc.api.command;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.METHOD)
public @interface Command {

    String name();

    String permission() default "";

    String noPermissionMessage() default "You do not have permission to run this command";

    String[] aliases() default {};

    String description() default "";

    String usage() default "";

    Class<? extends CommandSender> allow() default CommandSender.class;

}

File: src/main/kotlin/org/fountainmc/api/command/Command.kt
package org.fountainmc.api.command

import kotlin.reflect.KClass

@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
annotation class Command(val name: String, val permission: String = "", val noPermissionMessage: String = "You do not have permission to run this command", val aliases: Array<String> = arrayOf(), val description: String = "", val usage: String = "", val allow: KClass<out CommandSender> = CommandSender::class)
